articles/how-to-fix-memory-leaks-in-python/index.md

```diff
@@ -0,0 +1,55 @@
+
+As a programmer, you may create large volumes of memory and then fail to release any. If your application uses more memory and doesn&#39;t release any, it exhausts the server&#39;s memory pool with time. That makes your application crash the next time it consumes more memory.
+
+Memory leaks were more prevalent when programmers only used C &amp; C++. This is because one was required to free memory from the application manually. More memory is used when an application is running and outdated data piles up in the register. Then, the program stops to run which purposely frees memory. However, the program may stop functioning if the application crashes.
+
+### What causes memory leaks in Python?
+
+[Python](https://www.python.org/) program, just like other programming languages, experiences memory leaks. Memory leaks in Python happen if the garbage collector doesn&#39;t clean and eliminate the unreferenced or unused data from Python.
+
+Python developers have tried to address memory leaks through the addition of features that free unused memory automatically. However, some unreferenced objects may pass through the garbage collector unharmed, resulting in memory leaks.
+
+Below are factors that cause memory leaks in Python.
+
+#### Lingering of large objects in the memory that aren&#39;t released
+
+Lingering objects occur when the domain controller can&#39;t replicate for a time interval longer than the [tombstone lifetime](https://support.storagecraft.com/s/article/Understanding-Tombstones-Active-Directory-and-How-To-Protect-It). The domain controller then reconnects to [replication topology](https://www.monitis.com/blog/active-directory-replication-topology/#). If you delete an object from the active directory service when the domain controller is offline, the object stays in the domain controller as a lingering object. It&#39;s those lingering objects that consume space leading to the occurrence of memory leaks.
+
+#### Reference styles in the code
+
+Referencing style will determine whether memory leaks will occur or will be avoided. A reference has an address and class information concerning objects being referenced. Assigning references doesn&#39;t create distinct duplicate objects. But if an object is no longer in use and can&#39;t be garbage collected because it&#39;s being referenced in another place within the application, it results in memory leaks. Various types of references are used in [code referencing](https://guides.libraries.uc.edu/citing/code), and they have different abilities to be garbage collected. A strong reference style is the most convenient to use in daily programming. But any object with a strong reference attached to it makes it hard for garbage collection. In such a case, when such objects pile up, they cause memory leaks.
```

Comment:
```suggestion
Referencing style will determine whether memory leaks will occur or will be avoided. A reference has an address and class information concerning objects being referenced. Assigning references doesn't create distinct duplicate objects. But if an object is no longer in use and can't be garbage collected because it's being referenced in another place within the application, it results in memory leaks. Various types of references are used in [code referencing](https://guides.libraries.uc.edu/citing/code), and they have different abilities to be garbage collected. A strong reference style is the most convenient to use in daily programming. But any object with a strong reference attached to it makes it hard for garbage collection. In such a case, when such objects pile up, they cause memory leaks.
```

Reply:
Please make these changes by replacing `&#39;` with ` ' ` throughout the article